### PR TITLE
Add segmentation pipeline widget

### PR DESCRIPTION
**Pull request by GwydionJon**
This solves #139

### CODE REVIEW

**Comment by dokempf** [comment]
The failing test is a change to a notebook that should not have been part of the PR in the first place.

File: adaptivefiltering/segmentation.py [hunk 1]
```diff
@@ -88,14 +88,30 @@ def save(self, filename):
...
-        :param grid:
-             The grid object which holds the map and the right side interface
-        :type grid: ipyleaflet.grid
+        :
         """
 
         segmentation_map = Map(segmentation=self)
         return segmentation_map.show()
 
+    def convert(self, srs_in, srs_out="EPSG:4326"):
```
Comment (dokempf) [inline adaptivefiltering/segmentation.py:97]: Having this as a user facing API suggests that there is a use case where a user needs this. When would that be?

File: adaptivefiltering/segmentation.py [hunk 2]
```diff
@@ -291,28 +323,71 @@ def setup_controls(self):
...
+
+    def load_segmentation(self, segmentation, override=False):
+        """Imports a segmentation object into the draw control data
+
+        :param segmentation:
+            A segmentation object which is to be loaded.
+        :type segmentation: Segmentation
+        """
+
+        if override == True:
```
Comment (dokempf) [inline adaptivefiltering/segmentation.py:369]: For booleans, just do `if override:`

**Review by dokempf — CHANGES_REQUESTED** [review]
I added some review comments. They mostly refer either to style or the change of user facing APIs. Once that is fixed, we need to do a second iteration of the feature after my discussion with Bernhard today.

File: adaptivefiltering/apps.py [hunk 3]
```diff
@@ -22,12 +22,14 @@
 import pyrsistent
 import pytools
 import wrapt
+from ipyleaflet import Marker
```
[the remark this answers is not in the document]
Comment (GwydionJon) [reply adaptivefiltering/apps.py:25]: Marker is no longer needed anyway.

File: adaptivefiltering/dataset.py [hunk 4]
```diff
@@ -185,23 +185,61 @@ def restrict(self, segmentation=None):
...
         """
-        create one or multiple segmentations from a dataset.
+
+        from adaptivefiltering.pdal import PDALInMemoryDataSet
+
+        dataset = PDALInMemoryDataSet.convert(self)
+
+        return dataset.crop_segmentation()
+
+    def assign_pipeline(self, segmentation, pipelines):
```
[the remark this answers is not in the document]
Comment (GwydionJon) [reply adaptivefiltering/dataset.py:204]: My thinking was that this makes it easier to associate the operation with the given dataset.

File: adaptivefiltering/segmentation.py (hunk 1 above)
Comment (dokempf) [reply adaptivefiltering/segmentation.py:97]: Okay, so we remove that.

File: adaptivefiltering/dataset.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dokempf) [reply adaptivefiltering/dataset.py:204]: I think I would rather not add more things to DataSet.

**Pull request merged**